Engine/DataFeeds/ZipDataCacheProvider.cs

```diff
@@ -307,6 +319,45 @@ private Stream CreateStream(CachedZipFile zipFile, string entryName, string file
             return null;
         }
 
+        /// <summary>
+        /// Cache a Zip
+        /// </summary>
+        /// <param name="filename">Zip to cache</param>
+        /// <param name="cachedZip">The resulting CachedZipFile</param>
+        /// <returns></returns>
+        private bool Cache(string filename, out CachedZipFile cachedZip)
+        {
+            cachedZip = null;
+            var dataStream = _dataProvider.Fetch(filename);
+            if (dataStream != null)
+            {
+                try
+                {
+                    cachedZip = new CachedZipFile(dataStream, DateTime.UtcNow, filename);
+
+                    if (!_zipFileCache.TryAdd(filename, cachedZip))
+                    {
+                        // some other thread could of added it already, lets dispose ours
+                        cachedZip.Dispose();
+                    }
+
+                    return true;
```

Comment:
Hm disposing of the cachedZip but returning true? Think this case should be handled differently  `some other thread could of added it already, lets dispose ours`, should probably fetch the cachedZip in the collection instead?